lib/utils/compose.coffee

```diff
@@ -349,13 +349,40 @@ tagServiceImages = (docker, images, serviceImages) ->
 			logs: d.logs
 			props: d.props
 
-authorizePush = (tokenAuthEndpoint, registry, images) ->
+
+getPreviousRepos = (logger, appID) ->
+	sdk = require('balena-sdk').fromSharedOptions()
+	sdk.pine.get(
+		resource: 'release'
+		options:
+			$filter:
+				belongs_to__application: appID
+				status: 'success'
+			$expand:
+				contains__image:
+					$expand: 'image'
+	)
+
+	.then (release) ->
+            # grab all images from the latest release, return all image locations in the registry
+		images = release[0].contains__image
+		repos = []
+		for i in images
+			imageName = i.image[0].is_stored_at__image_location
+			[ _, _, repoName, _ = 'latest' ] = /(.*?)\/(.*?)(?::([^/]*))?$/.exec(imageName)
```

Comment:
Might be better to use a tighter regex, as in https://github.com/balena-io-modules/docker-toolbelt/blob/master/lib/docker-toolbelt.coffee#L338 (you could even use that docker-toolbelt module)

Reply:
Admittedly I stole this regex from https://github.com/balena-io/balena-cli/blob/master/lib/utils/compose.coffee#L338, but I think making use of docker-toolbelt sounds like a much better long-term plan. Will update that, thank you!